docs/specs/target.md

```diff
@@ -0,0 +1,157 @@
+# Target Specification
+
+This document specifies requirements for deployment targets for the
```

Comment:
Broadly we can split our targets between `Virtual/Physical Machines` and `Orchestrators`.

- `Virtual/Physical Machines`
  - Target is a single instance.
  - This implies limited, to no, _High Availability_ and _Scaling_ spec adherence.
  - However, there should be no issues with general _Architecture_, _Sizing_, and _Hardening_.
- `Orchestrators`
  - Target is a group of nodes, managed by a scheduler or orchestrator.
  - A management layer (Kubernetes, Nomad, Swarm, Mesos, DC/OS) over a group of nodes allows us to adhere to _High Availability_ and _Scaling_ spec more easily.
  - These targets should expose primitives to ensure the _Architecture_, _Sizing_, and _Hardening_ spec is met.

There is some space between these two classes for things like Chef, Puppet, Ansible, or even Terraform that could help bridge this gap but we do not currently provide any resources for these types of deployments.